internal/pkg/agent/cmd/upgrade.go

```diff
@@ -94,3 +100,16 @@ func upgradeCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error
 	fmt.Fprintf(streams.Out, "Upgrade triggered to version %s, Elastic Agent is currently restarting\n", version)
 	return nil
 }
+
+func checkValidUri(rawURI string) error {
+	uri, err := url.Parse(rawURI)
+	if err != nil {
+		return err
+	}
+
+	if !strings.EqualFold(uri.Scheme, "https") {
```

Comment:
Do we also want to add a similar change for upgrade actions?
https://github.com/elastic/elastic-agent/blob/main/internal/pkg/agent/application/upgrade/artifact/download/verifier.go#L179

Reply:
yes we could. good point